public/src/components/PostComponent.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+    faAngleUp,
+    faAngleDown,
+    faCommentAlt,
+    faShare,
+    faBookmark,
+    faStar,
+} from '@fortawesome/free-solid-svg-icons';
+import methods from '../utils/methods';
+
+const {
+    getDate,
+    reformatTextToHtml,
+} = methods;
+
+export default class PostComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.calculateDate = this.calculateDate.bind(this);
+    }
+
+    getPostImage() {
+        const { thumbnail, thumbnail_height, thumbnail_width } = this.props;
+        const imageWidth = thumbnail_width !== 'null' ? +thumbnail_width : 'auto';
+        const imageHeight = thumbnail_height !== 'null' ? +thumbnail_height : 'auto';
+
+        return thumbnail && thumbnail !== 'null' && thumbnail !== 'self'
```

Comment:
`null` and `self` are kind of confusing here, maybe use some self-descriptive variable like

```
const REDDIT_THUMBNAIL_STATE = {
    SELF: 'self',
    NULL: 'null'
}
```

Reply:
fixed https://github.com/YuriyBC/Reddit-archive/commit/5b4eb5665ecbbcf1c9d0a151dafe2e226dd4a075